### DISCUSSION TITLE
Customize FIT of individual wallpapers

### DISCUSSION THREAD

**Discussion by TerminatedProcess** [rocksdanister/lively#470] [Ideas]
Myself, I like to take videos with my phone and then play them in Lively. Eg a field of waving grass or stream.. my dogs.. some of the videos when imported need a different fit to work (based on how I hold my phone when I create the video). We have a FIT setting in settings. Can that be the default but also each wallpaper would have an independent FIT setting under Customize? Then if needed, you can adjust the FIT for just that wallpaper. Good idea? Bad? It's not a huge deal but I thought it might be a nice feature. Beautiful application!

**Comment by TerminatedProcess** [comment]
Bad fit example..

![image](https://user-images.githubusercontent.com/63432380/116800060-a3e97080-aacb-11eb-99ad-4364c6ffd32e.png)

**Comment by rocksdanister** [comment]
> Beautiful application!

Thanks!

> Can that be the default but also each wallpaper would have an independent FIT setting under Customize?

Yeah I am thinking about it, the issue is..
The customise menu is based on [LivelyProperties API](https://github.com/rocksdanister/lively/wiki/Web-Guide-IV-:-Interaction#lively-properties) which is a general way of writing ui controls, it is not aware of what exactly the each control does.

Originally LivelyProperties was made for webpage wallpapers, I have since then modified and added support for mpv player by including a [default property file](https://github.com/rocksdanister/lively/wiki/Video-Guide#mpv) ; although I did not document it and some syntax are different from webpage version, users can write their own mpv property file by following the [mpv manual](https://mpv.io/manual/master/) and adding the LivelyProperties.json file in the media folder and hitting refresh!

Back to the point, with the current implementation even if I write a Fit/scaler property in LivelyProperties.json file - Lively will not be aware of what exactly its doing and this means I can't update it when you change the fit settings via main app (which is not part of LivelyProperties.)

Need to think of a good solution..

**Reply by TerminatedProcess** [reply]
I didn't realize it would be that complex! It's not a show stopper or anything though..

**Reply by TerminatedProcess** [reply]
Maybe you need an AI kind of setting? It will always try to do the best fit on it's own. Then the problem becomes about playing instead of the coming up with a different settings architecture. In case you are interested.. I use an android app on my phone called Animgif. It's purpose is to play gif files from a folder (randomly if you like). It does have a best fit setting. I'm mentioning to just demonstrate the idea.. plus it's a fun app. My phone is interesting because of it.. lol

**Reply by rocksdanister** [reply]
and then block chain and machine learning 😅
I think I will just make a few new special properties.. similar to the `Restore default` button that gets auto added.

Without AI I think many video players have settings like best fit/fit from outside that does this, it might be [in the docs](https://mpv.io/manual/master/) somewhere.

**Reply by TerminatedProcess** [reply]
haha, I probably shouldn't have said AI, you had too much fun with it!

**Reply by rocksdanister** [reply]
I think this is what you are looking for [--no-keepaspect-window](https://mpv.io/manual/master/#options-no-keepaspect-window)

**Reply by rocksdanister** [reply]
@TerminatedProcess 

https://user-images.githubusercontent.com/17554161/117413402-8edd6900-af33-11eb-9325-31b97f9b4ea1.mp4

**Reply by TerminatedProcess** [reply]
Maybe I'm not using it right.. I downloaded the latest mpv and created a %localappdata%/lively windows/mpv folder and put it in there. Then added a mpv.conf file and put inside of it a single entry with 

no-keepaspect-window

I also tried making a .conf file using the name of the video file.. but no luck? What am I missing?

**Reply by rocksdanister** [reply]
Actually is this not similar to Uniform scaling xD
